syntax = "proto3";
package plugin_IPC.toPlugin;

message Information {
	string type = 1;
	string version = 2;
	string author = 3;
	string url = 4;
	string identity = 5;
	string toString = 6;
	bool supportedOnThisPlatform = 7;
	bool supportManuallyCreatedDevice = 8;
	string packageFileContent = 9;
	string path = 10;
	bool supportDeviceRemovedNotification = 11;
}

message ProxySettings {
	string host = 1;
	uint32 port = 2;
	string username = 3;
	string password = 4;
	string bypassRegex = 5;
}

message Init {
	Information pluginInformation = 1;
	string dataPath = 2;
	string logFile = 3;
	string logLevel = 4;
	ProxySettings proxySettings = 5;
}

